aria_data_loaders/aria_data_utils/aria_streamer.py

```diff
@@ -0,0 +1,1101 @@
+        and return a tuple of (valid, score, result_image)
+        Input:
+        - img: np.ndarray: image to run inference on
+        - heuristic: function: function to score the detections
+        Output:
+        - valid: bool: whether the frame is valid or not
+        - score: float: score of the frame (based on :heuristic:)
+        - result_image: np.ndarray: image with bounding boxes drawn on it
+        """
+        detections, result_image = self.object_detector.run_inference_and_return_img(
+            img
+        )
+        if heuristic is not None:
+            valid, score, stop = heuristic(detections)
+
+        return valid, score, stop, result_image
+
+    def _p1_demo_heuristics(
+        self, detections, img_size=(512, 512)
+    ) -> Tuple[bool, Dict[str, bool], Dict[str, float]]:
+        """*P1 Demo Specific*
+        Heuristics to filter out unwanted and bad obejct detections. This heuristic
+        scores each frame based on (a) % of pixels occupied by object in the frame
+        and (b) proximity of bbox to image center. Further only those images are
+        scored which have only the object detection without a hand in the frame.
+
+        Detections are expected to have the format:
+        [["object_name", "confidence", "bbox"]]
+        """
+        valid = False
+        score = {}
+        stop = {}
+        # figure out if this is an interaction frame
+        if len(detections) > 1:
+            objects_in_frame = [det[0] for det in detections]
+            # FIXME: only works for 1 object of interest right now, extend to multiple
+            if "hand" in objects_in_frame and "water bottle" in objects_in_frame:
+                print(f"checking for intersection b/w: {object}")
+                stop["water bottle"] = check_bbox_intersection(
+                    detections[objects_in_frame.index("water bottle")][2],
+                    detections[objects_in_frame.index("hand")][2],
+                )
+                print(f"Intersection: {stop}")
+        if len(detections) == 1:
+            if "water bottle" == detections[0][0]:
+                score["water bottle"] = centered_heuristic(detections)[0]
+                valid = True
+
+        return valid, score, stop
+
+    def get_vrs_timestamp_from_img_idx(
+        self, stream_name: str = STREAM1_NAME, idx_of_interest: int = -1
+    ):
+        """
+        Get VRS timestamp corresponding to VRS image index
+
+        Args:
+            stream_name (str, optional): Stream name. Defaults to STREAM1_NAME.
+            idx_of_interest (int, optional): Index of interest. Defaults to -1 i.e. the last position of VRS.
+
+        Returns:
+            int: Corresponding VRS timestamp in nanoseconds
+        """
+        stream_id = self.provider.get_stream_id_from_label(stream_name)
+        frame_data = self.provider.get_image_data_by_index(stream_id, idx_of_interest)
+        return frame_data[1].capture_timestamp_ns
+
+    def parse_camera_stream(
```

Comment:
Split this up into two, one function only visualizes stream. Other uses classes to "process" the frame.

Reply:
As discussed, pushing this as a TODO for next PR. Added a TODO in the code.